docs/rest_api.rst

```diff
@@ -0,0 +1,951 @@
+
+Get Item
+--------
+
+The get item endpoint is very simple, and was already covered to some extent.
+The response data structure is::
+
+    {
+        "id": "<Primary Key>"
+        "description_columnns": {},
+        "label_columns": {},
+        "include_columns": [],
+        "result": {}
+    }
+
+Now we are going to cover the *Rison* arguments for custom fetching
+meta data keys or columns. This time the accepted arguments is slightly extended::
+
+    {
+        "keys": [ ... List of meta data keys to return ... ],
+        "columns": [ ... List of columns to return ... ]
+    }
+
+So for fetching only the *name* and *address* for a certain *Contact*, using *Rison*::
+
+    (columns:!(name,address))
+
+Our *curl* command will look like::
+
+    curl 'http://localhost:8080/api/v1/contact/1?q=(columns:!(name,address))' \
+    -H "Content-Type: application/json" \
+    -H "Authorization: Bearer $TOKEN"
+    {
+      "description_columns": {},
+      "id": "1",
+      "include_columns": [
+        "name",
+        "address"
+      ],
+      "label_columns": {
+        "address": "Address",
+        "name": "Name"
+      },
+      "result": {
+        "address": "Street phoung",
+        "name": "Wilko Kamboh"
+      }
+    }
+
+And to only include the *label_columns* meta data, *Rison* data structure::
+
+    (columns:!(name,address),keys:!(label_columns))
+
+Our *curl* command will look like::
+
+    curl 'http://localhost:8080/api/v1/contact/1?q=(columns:!(name,address),keys:!(label_columns))' \
+    -H "Content-Type: application/json" \
+    -H "Authorization: Bearer $TOKEN"
+    {
+      "id": "1",
+      "label_columns": {
+        "address": "Address",
+        "name": "Name"
+      },
+      "result": {
+        "address": "Street phoung",
+        "name": "Wilko Kamboh"
+      }
+    }
+
+We can restrict or add fields for the get item endpoint using
+the ``show_columns`` property. This takes precedence from the *Rison* arguments::
+
+    class ContactModelRestApi(ModelRestApi):
+        resource_name = 'contact'
+        datamodel = SQLAInterface(Contact)
+        show_columns = ['name']
+
+We can add fields that are python functions also, for this on the SQLAlchemy definition,
+let's add a new function::
+
+    class Contact(Model):
+        id = Column(Integer, primary_key=True)
+        name = Column(String(150), unique=True, nullable=False)
+        address = Column(String(564))
+        birthday = Column(Date, nullable=True)
+        personal_phone = Column(String(20))
+        personal_celphone = Column(String(20))
+        contact_group_id = Column(Integer, ForeignKey('contact_group.id'), nullable=False)
+        contact_group = relationship("ContactGroup")
+        gender_id = Column(Integer, ForeignKey('gender.id'), nullable=False)
+        gender = relationship("Gender")
+
+        def __repr__(self):
+            return self.name
+
+        def some_function(self):
```

Comment:
Logically this can only work for arg-free functions. Maybe the implementation should force `@property` decorated methods only (?)

Reply:
That's a good idea! I wonder if partials could make sense here, if not I'll force it